Introduced additional dependencies in the testsuite

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

testsuite

### Describe the bug

We've encountered particular issues while executing tests from the `integration-arquillian/` module.

```
12:49:23,604 INFO  [org.keycloak.testsuite.arquillian.undertow.KeycloakOnUndertow] Starting auth server on embedded Undertow on: http://0.0.0.0:8180
12:49:23,873 INFO  [io.undertow] starting server: Undertow - 2.2.5.Final
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.006 s <<< FAILURE! - in org.keycloak.testsuite.account.AccountRestServiceCorsTest
[ERROR] org.keycloak.testsuite.account.AccountRestServiceCorsTest  Time elapsed: 5.936 s  <<< ERROR!
java.lang.AbstractMethodError: Receiver class org.keycloak.adapters.elytron.KeycloakServletExtension does not define or inherit an implementation of the resolved method 'abstract void handleDeployment(io.undertow.servlet.api.DeploymentInfo, jakarta.servlet.ServletContext)' of interface io.undertow.servlet.ServletExtension.
```

There were introduced new unwanted dependencies into the testsuite, which altered the test execution from that module. For more specific sub-modules such as `tests` and `tests/base`, everything works as expected. 

You can see also in the log above that the Undertow version 2.2.5.Final is different than the one specified in the poms.

### Version

999

### Expected behavior

Parent and child POMs aligned with each other

### Actual behavior

Parent and child POMs are NOT aligned with each other

### How to Reproduce?

Execute test from the `/integration-arquillian` module

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/servers/app-server/jboss/pom.xml
+++ b/testsuite/integration-arquillian/servers/app-server/jboss/pom.xml
@@ -71,17 +71,30 @@
                 </file>
             </activation>
             <dependencies>
+                <!-- These ZIPs can be fetched from artifactItems, but we need to properly assess the Maven build execution order -->
                 <dependency>
                     <groupId>org.keycloak</groupId>
                     <artifactId>${app.server.oidc.adapter.artifactId}</artifactId>
                     <version>${oidc-adapter.version}</version>
                     <type>zip</type>
+                    <exclusions>
+                        <exclusion>
+                            <groupId>*</groupId>
+                            <artifactId>*</artifactId>
+                        </exclusion>
+                    </exclusions>
                 </dependency>
                 <dependency>
                     <groupId>org.keycloak</groupId>
                     <artifactId>${app.server.saml.adapter.artifactId}</artifactId>
                     <version>${saml-adapter.version}</version>
                     <type>zip</type>
+                    <exclusions>
+                        <exclusion>
+                            <groupId>*</groupId>
+                            <artifactId>*</artifactId>
+                        </exclusion>
+                    </exclusions>
                 </dependency>
             </dependencies>
             <build>

